### PR TITLE
Added input data provider: root process name

### PR DESCRIPTION
**Pull request by ajzach**

### CODE REVIEW

File: src/input/rpname.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	if (parent == ROOT_PID) {
+		name = read_proc_property(pid, PROC_NAME);
+		if (name != NULL)
+			return snprintf(input, SNOOPY_INPUT_MESSAGE_MAX_SIZE, "%s", name);
+		else
+			return snprintf(input, SNOOPY_INPUT_MESSAGE_MAX_SIZE, "%s",	UNKNOWN_STR);
+	} else if (parent == EMPTY_PID)
+		return snprintf(input, SNOOPY_INPUT_MESSAGE_MAX_SIZE, "%s", UNKNOWN_STR);
+	else {
+		return get_rpname(get_parent_pid(pid), input);
```
Comment (bostjan) [inline src/input/rpname.c]: You have "parent" variable which already contains parent pid, no need to call get_parent_pid() again, right?

**Comment by bostjan** [comment]
Also, could you please update header comments to reflect year 2015 and accurate file name?

And put your email in those new files, unless you are transferring ownership to me. :)

b.

**Comment by bostjan** [comment]
One more question: what is your actual use case that led you to development of this input provider? I know the answer "I wanted to log the root process name", the thing I am curious about is "why"?

Anyhow, I am just curious, no need to elaborate a lot here.

b.

**Comment by ajzach** [comment]
I want to distinguish commands that are executed via ssh (root process sshd).

**Pull request closed without merging**